PaddleAudio/examples/dcase2021_task1b/train.py

```diff
@@ -0,0 +1,214 @@
+
+    else:
+        best_acc = 0.0
+
+    step = 0
+    for epoch in range(start_epoch, epoch_num):
+
+        avg_loss = 0.0
+        avg_acc = 0.0
+        model.train()
+        model.clear_gradients()
+        t0 = time.time()
+        for batch_id, (xd, yd, p) in enumerate(train_loader()):
+
+            if args.task_type == 'audio_only':
+                p = None  # do not use probability from video branch
+            if step < warm_steps:
+                optimizer.set_lr(lrs[step])
+
+            xd.stop_gradient = False
+            if c['balanced_sampling']:
+                xd = xd.squeeze()
+                yd = yd.squeeze()
+            xd = xd.unsqueeze((1))
+            if c['mixup']:
+                mixed_x, mixed_p, mixed_y = mixup_data(xd, yd, p,
+                                                       c['mixup_alpha'])
+                logits = model(mixed_x, mixed_p)
+                pred = F.log_softmax(logits)
+                loss_val = mixup_loss(pred, mixed_y)
+                loss_val.backward()
+            else:
+                logits = model(xd, p)
+                pred = F.log_softmax(logits)
+                loss_val = nll_loss(pred, yd)
+                loss_val.backward()
+            optimizer.step()
+            model.clear_gradients()
+
+            acc = np.mean(np.argmax(pred.numpy(), axis=1) == yd.numpy())
+            avg_loss = (avg_loss * batch_id + loss_val.numpy()[0]) / (1 +
+                                                                      batch_id)
+            avg_acc = (avg_acc * batch_id + acc) / (1 + batch_id)
+
+            elapsed = (time.time() - t0) / 3600
+            remain = elapsed / (1 + batch_id) * (len(train_loader) - batch_id)
+
+            msg = f'epoch:{epoch}, batch:{batch_id}'
+            msg += f'|{len(train_loader)}'
+            msg += f',loss:{avg_loss:.3}'
+            msg += f',acc:{avg_acc:.3}'
+            msg += f',lr:{optimizer.get_lr():.2}'
+            msg += f',elapsed:{elapsed:.1}h'
+            msg += f',remained:{remain:.1}h'
+
+            if batch_id % 50 == 0 and local_rank == 0:
+                logger.info(msg)
+                log_writer.add_scalar(tag="train loss",
+                                      step=step,
+                                      value=avg_loss)
+                log_writer.add_scalar(tag="train acc", step=step, value=avg_acc)
+
+            step += 1
+            if step % c['checkpoint_step'] == 0 and local_rank == 0:
+
+                val_loss, val_acc = evaluate(epoch, val_loader, model, nll_loss,
+                                             args.task_type)
+                log_writer.add_scalar(tag="eval acc", step=epoch, value=val_acc)
+                log_writer.add_scalar(tag="eval loss",
+                                      step=epoch,
+                                      value=val_loss)
+
+                model.train()
+                model.clear_gradients()
```

Comment:
`clear_grad()` is a recommended usage for paddlepaddle 2.0

Reply:
fixed